src/lib/storage/ConnectionRecord.ts

```diff
@@ -34,4 +50,23 @@ export class ConnectionRecord extends BaseRecord implements ConnectionStoragePro
     this.endpoint = props.endpoint;
     this.tags = props.tags as { [keys: string]: string };
   }
+
+  get myKey() {
+    if (!this.didDoc) {
+      return null;
+    }
+    return this.didDoc.service[0].recipientKeys[0];
+  }
+
+  get theirKey() {
+    if (!this.theirDidDoc) {
+      return null;
+    }
+    return this.theirDidDoc.service[0].recipientKeys[0];
+  }
```

Comment:
Currently all connections use the first key with the first service so it should work fine, but i think it would be better to look for the `IndyAgent` service.

Reply:
I agree it's really not a robust solution :) It's implemented in the simplest possible way. I suggest to do it in other PR because it's not part of this change so much. Contribution welcome here ;)

I'm not sure if such logic should be part of the `ConnectionRecord`, maybe it'll be more complicated at the end. To be honest, I don't know what's the 100 % correct implementation.

Reply:
Not sure either. Maybe we can create a diddoc class as implemented here: https://github.com/decentralized-identity/did-common-typescript/blob/master/lib/DidDocument.ts

Getting my key vs their key would then be something like. 
```ts
const myKey = connection.didDoc.getServiceByType('IndyAgent').recipientKeys[0]
const theirKey = connection.theirDidDoc.getServiceByType('IndyAgent').recipientKeys[0]
```

But i agree this should be addressed in a separate PR.

Reply:
Cool, I like that idea 👍 I didn't know about that repo, thanks for pointing that out. We can take it as an inspiration, or maybe, add the whole package as a dependency. Although, I'm not sure about the work needed for the persistence of such an object with our current storage mechanism.

Reply:
Created #53